test/support/publish_dsl.rb

```diff
@@ -25,11 +25,191 @@ def publish_wait
+        puts "Episode ID: #{ep_id}" if debug?
+        del_url = 'UNKNOWN'
+        authz_token = 'UNKNOWN'
+        page.driver.network_traffic.each do |req|
+          if req.url.match(/authorization$/)
+            del_url = req.url + "/stories/#{ep_id}"
+            req.headers.each do |h|
+              if h['name'] == 'Authorization'
+                authz_token = h['value']
+                break
+              end
+            end
+          end
+          break if authz_token != 'UNKNOWN'
+        end
+        puts "DELETE #{del_url}" if debug?
+        page.execute_script(publish_js_to_delete_episode(del_url, authz_token))
+      end
+      sleep(1) # time to finish XHR
+    end
+
+    def publish_js_to_delete_episode(del_url, authz_token)
+      debugger_line = debug? ? "console.log('DELETE ok')" : ''
+      <<~JSCRIPT
+        var xhr=new XMLHttpRequest();
+        xhr.open('DELETE','#{del_url}');
+        xhr.setRequestHeader('Authorization', '#{authz_token}');
+        xhr.setRequestHeader('Accept', 'application/hal+json');
+        xhr.onload=function() {
+          if (xhr.status === 204) {
+            #{debugger_line}
+          } else {
+            console.log('XHR fail:' + xhr.status)
+          }
+        };
+        xhr.withCredentials = true;
+        xhr.send();
+      JSCRIPT
+    end
+
+    def create_series!(random_str = SecureRandom.hex(10))
+      publish_login!
+      visit CONFIG.PUBLISH_HOST
+      click_link 'New Series'
+      fill_in 'Series Title', with: "Test Series #{random_str}"
+      fill_in 'Teaser', with: "Test Series Teaser #{random_str}"
+      click_button 'Create'
+      publish_wait_for_success
+      publish_wait_for_title
+      page.must_have_content("Test Series #{random_str}")
+      page.must_have_content('Podcast Info')
+
+      publish_wait_for_episodes_link
+      page.must_have_content("0 Episodes in Series")
+
+      click_link 'Podcast Info'
+      click_button 'Create Podcast'
+      # TODO how to select dropdown checkbox?
+      # find #category.dropdown-toggle, click, then select
+      current_url
+    end
+
+    def set_upload_input_file(selector, path)
+      # can't use attach_file() because our input field is invisible
+      Capybara.ignore_hidden_elements = false
+      upload_input = find(selector)
+      upload_input.set(path)
+      Capybara.ignore_hidden_elements = true
+    end
+
+    def publish_wait_for_duration
+      find('dt', text: 'Duration:')
+    end
+
+    def publish_wait_for_save_on_upload
+      find_button('Save', wait: 10)
+    end
+
+    def publish_wait_for_publish_button
+      find_button('Publish', wait: 30)
+      find('p', text: 'Ready to publish', wait: 10)
+    end
+
+    def create_episode!(series_url, random_str = SecureRandom.hex(10))
+      visit series_url
+      click_link '0 Episodes in Series' # TODO better way to trigger create link?
+
+      # TODO why doesn't click_link work here? fetching the href attribute and visit() does...
```

Comment:
🤔 you got me, works in the browser

Reply:
Masked by the image/icon or something?  But that doesn't make sense either.

Reply:
I think it might have to do with the routing? Does publish load the page in-place rather than do a full page request/response cycle?

Reply:
yep, it's routed through the SPA. When I looked at this in Chrome, I just did a findElement and a click() on it and it navigated fine.

Reply:
I think there's something weird about how poltergeist handles those SPA actions then. Not sure what, just that if I did a full `visit` rather than just following the link clicks, it worked.

Reply:
It would spit out any js errors, right?  (A javascript error would certainly break SPA navigation).